virtualizarr/writers/icechunk.py

```diff
@@ -0,0 +1,179 @@
+from typing import TYPE_CHECKING, cast
+
+import numpy as np
+from xarray import Dataset
+from xarray.backends.zarr import encode_zarr_attr_value
+from xarray.core.variable import Variable
+from zarr import Group
+
+from virtualizarr.manifests import ChunkManifest, ManifestArray
+from virtualizarr.zarr import encode_dtype
+
+if TYPE_CHECKING:
+    from icechunk import IcechunkStore
+
+
+VALID_URI_PREFIXES = {
+    "s3://",
+    # "gs://",
+    # "azure://",
+    # "r2://",
+    # "cos://",
+    # "minio://",
+    "file:///",
+}
+
+
+def dataset_to_icechunk(ds: Dataset, store: "IcechunkStore") -> None:
+    """
+    Write an xarray dataset whose variables wrap ManifestArrays to an Icechunk store.
+
+    Currently requires all variables to be backed by ManifestArray objects.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+    store: IcechunkStore
+    """
+    from icechunk import IcechunkStore
```

Comment:
I wonder if we should add a warning here that points people to the Earthmover docs describing all the dependencies that are required